src/engine/enginebuffer.cpp

```diff
@@ -1191,21 +1200,23 @@ void EngineBuffer::processSeek(bool paused) {
     }
 
     if (!paused && (seekType & SEEK_PHASE)) {
-        if (kLogger.traceEnabled()) {
+        if (kLogger.traceEnabled())
             kLogger.trace() << "EngineBuffer::processSeek Seeking phase";
-        }
         double requestedPosition = position;
         double syncPosition = m_pBpmControl->getBeatMatchPosition(position, true, true);
         position = m_pLoopingControl->getSyncPositionInsideLoop(requestedPosition, syncPosition);
-        if (kLogger.traceEnabled()) {
+        if (kLogger.traceEnabled())
             kLogger.trace()
                     << "EngineBuffer::processSeek seek info: " << m_filepos_play
                     << " -> " << position;
-        }
     }
     if (position != m_filepos_play) {
-        if (kLogger.traceEnabled()) {
+        if (kLogger.traceEnabled())
             kLogger.trace() << "EngineBuffer::processSeek Seek to" << position;
+        if (m_bHotcueJumpPending) {
+            m_bHotcueJumpPending = false;
+            qDebug() << "Storing jump to Macro";
+            m_pEngineMaster->m_pMacroRecording->appendHotcueJump(m_filepos_play, position);
```

Comment:
Check if ptr is null.

Reply:
If we properly track the state, this can never be null at this point.
I'll add a DEBUG_ASSERT.

Reply:
Use VERIFY_OR_DEBUG_ASSERT to fail gracefully in release builds even if the pointer is null.

Reply:
I don't quite get the point. With proper real-time guards, this case is impossible. And how do I "fail gracefully" at that point? It's not like I can break an if, right?

Reply:
Yes, it should be impossible. But if somehow it does happen because of a bug, Mixxx shouldn't crash.

Reply:
Fixed :)